tiledb/sm/cpp_api/config.h

```diff
@@ -49,10 +49,13 @@ class Config;  // Forward decl for impl classes
 
 namespace impl {
 
-class ConfigIter : public std::iterator<
-                       std::forward_iterator_tag,
-                       const std::pair<std::string, std::string>> {
+class ConfigIter {
```

Comment:
@dhoke4tdb is this change still desired? Do we want to take the iterator out of the class? I don't think this is relevant to the PR right? If you agree I'll restore this class to the current dev version.

Reply:
@Shelnutt2 
std::iterator is in some form deprecated in c++17, but because we have deprecation disabled in our builds it is not reported.
It's not taken out perse, there are using statements below where you commented that continue support for iteration.
So what I have, or something else, is likely going to eventually be needed, but if you don't want it hidden in this that's fine.  I think I recently encountered another recently added iterator somewhere that encounters the same warning.

Reply:
@dhoke4tdb thanks for the details. I went ahead and removed it from this PR and opened a story about the issue for followup.